File: WoofAndroid/presentation/profile/src/main/kotlin/com/perrystreet/woof/presentation/profile/ui/extensions/ProfileErrorToToastMapper.kt
package com.perrystreet.woof.presentation.profile.ui.extensions

import com.perrystreet.woof.models.errors.MessageException
import com.perrystreet.woof.models.errors.WoofException
import com.perrystreet.woof.presentation.common.error.ErrorToast
import com.perrystreet.woof.presentation.common.error.IErrorToToastMapper
import com.perrystreet.woof.resources.R

class ProfileErrorToToastMapper(
    private val name: String,
) : IErrorToToastMapper {
    override fun invoke(error: Throwable): ErrorToast? = when (error) {
        WoofException.AlreadyWoofed -> ErrorToast(R.string.profile_toast_already_woofed, listOf(name))
        MessageException.EmptyMessage -> ErrorToast(R.string.profile_toast_message_empty)
        else -> null
    }
}

File: WoofIOS/SwiftPackages/Presentation/Profile/Sources/PresentationProfile/UI/Extensions/ProfileErrorToToastMapper.swift
import Models
import PresentationCommon
import Resources

struct ProfileErrorToToastMapper: ErrorToToastMapping {
    let name: String

    func callAsFunction(_ error: Swift.Error) -> ErrorToast? {
        switch error {
        case WoofError.alreadyWoofed: ErrorToast(message: L10n.Profile.toastAlreadyWoofed(name))
        case MessageError.emptyMessage: ErrorToast(message: L10n.Profile.toastMessageEmpty)
        default: nil
        }
    }
}
